libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -64,44 +64,52 @@ bool NameAndTypeResolver::resolveNamesAndTypes(ContractDefinition& _contract)
 	{
 		m_currentScope = &m_scopes[nullptr];
 
+		ReferencesResolver resolver(m_errors, *this, &_contract, nullptr);
+		bool result = true;
 		for (ASTPointer<InheritanceSpecifier> const& baseContract: _contract.baseContracts())
-			ReferencesResolver resolver(*baseContract, *this, &_contract, nullptr);
+			result = resolver.resolve(*baseContract);
```

Comment:
This will overwrite `result` in the second iteration.
Please use something like
```if (!resolver.resolve(*baseContract))
    result = false;```
In most cases we can continue calling `resolver.resolve` even on error.